luigi/contrib/pyspark_runner.py

```diff
@@ -51,14 +106,32 @@ def __init__(self, job, *args):
         self.args = args
 
     def run(self):
-        from pyspark import SparkContext, SparkConf
+        from pyspark import SparkConf
         conf = SparkConf()
         self.job.setup(conf)
-        with SparkContext(conf=conf) as sc:
+        with self._entry_point_class(conf) as (entry_point, sc):
             self.job.setup_remote(sc)
-            self.job.main(sc, *self.args)
+            self.job.main(entry_point, *self.args)
+
+
+def _pyspark_runner_with(name, entry_point_class):
+    return type(name, (AbstractPySparkRunner,), {'_entry_point_class': entry_point_class})
+
+
+PySparkRunner = _pyspark_runner_with('PySparkRunner', SparkContextEntryPoint)
+PySparkSessionRunner = _pyspark_runner_with('PySparkSessionRunner', SparkSessionEntryPoint)
+
+
+def _use_spark_session():
+    return bool(configuration.get_config().get('pyspark_runner', "use_spark_session", False))
```

Comment:
Is this a new config class section? If so, how would you support the ability for users to decide on their own section name?

Reply:
@dlstadther 
I didn't get it
Could you please explain?

Reply:
Nevermind here. I looked through some other examples and we don't generally allow configurability of section names.